seismostats/analysis/avalue/tests/test_avalue.py

```diff
@@ -37,21 +38,21 @@ def test_estimate_a():
 def test_estimate_a_classic():
     mags = np.array([1, 2, 3, 4, 5, 6, 7, 8, 9, 10])
     estimator = ClassicAValueEstimator()
-    a_estimate = estimator.calculate(mags, mc=1, delta_m=10.0)
+    a_estimate = estimator.calculate(mags, mc=1, delta_m=1)
     assert a_estimate == 1.0
 
     # reference magnitude is given and b-value given
-    a_estimate = estimator.calculate(mags, mc=1, delta_m=0.0,
-                                     m_ref=0, b_value=1)
+    with warnings.catch_warnings(record=True):
+        a_estimate = estimator.calculate(mags, mc=1, delta_m=0.0,
+                                         m_ref=0, b_value=1)
```

Comment:
why catch a warning?

Reply:
because is makes the pytest go all yellow, I was not sure if that was a problem. Also, its annoying that there is a warning even if everything works as it should

Reply:
then maybe there should be no warning if everything works as it should?

Reply:
No, the warning is correct, it tells us that it does not check delta_m=0 but only up to the precision 1e-15.
Maybe a way around this would be to not use delta_m = 0.. what do you think?

Reply:
Okay, let's see, I have just skimmed over the related code to that warning, so I may be wrong:

If delta_m=0, it tells you that it is smaller than precision and checks for delta_m=precision.

- btw. if I am correct, it tells you that delta_x is smaller than tolerance, which is confusing to the user since he entered delta_m?

But basically, it will now either pass the test, whereas imo. the warning was not required, or it will fail the test, which is imo. the only case when the warning is relevant.

It is very possible that I missunderstood something?

Reply:
Hm, I think you got it right:
- The binning test warns you if you gave it a delta_x smaller than the precision
- Yes, the warning for the b-value estimator user will probably be confusing.. I will think of a solution

Reply:
Ok, now, there is no warning in case of delta_m = 0- I just directly input it in the binning function with 1e=8 (which is so small that a difference in binning would really make no difference in any result)

Reply:
Actually I do agree!
So now, I changed the binning function, such that it does not give a warning if delta_x is smaller than the tolerance.